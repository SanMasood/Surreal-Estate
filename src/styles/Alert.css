.Alert{
    box-sizing: border-box;
    
}

.Alert.alert-success{
    border: 2px solid green;
}

.Alert.alert-error{
    border: 2px solid red;
}