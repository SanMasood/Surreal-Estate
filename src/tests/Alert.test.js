import React from 'react';
import { render } from '@testing-library/react';
import Alert from '../components/Alert';

test('displays error message', () => {
  const { getByText } = render(<Alert message="Error!" />);
  // const alertMessageNode = component.getByText('Error!');

  expect(getByText(/Error/).textContent).toBe('Error!');
});

test('displays success message', () => {
  const { getByText } = render(<Alert message="Success!" />);

  expect(getByText(/success/i).textContent).toBe('Success!');
});

test('does not render error or success if message props is empty', () => {
  const { asFragment } = render(<Alert message="" />);
  expect({ asFragment }).toMatchSnapshot();
});

test('renders error message if property not created', () => {
  const { getByText, asFragment } = render(<Alert message="Error!" />);
  expect(asFragment()).toMatchSnapshot();
});

test('renders success message if property created', () => {
  const { getByText, asFragment }=render(<Alert message='Success!'/>);
  expect(asFragment()).toMatchSnapshot();
})