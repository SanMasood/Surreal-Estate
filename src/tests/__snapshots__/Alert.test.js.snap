// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not render error or success if message props is empty 1`] = `
Object {
  "asFragment": [Function],
}
`;

exports[`renders error message if property not created 1`] = `
<DocumentFragment>
  <div
    class="Alert alert-error"
  >
    Error!
  </div>
</DocumentFragment>
`;

exports[`renders success message if property created 1`] = `
<DocumentFragment>
  <div
    class="Alert alert-error"
  >
    Success!
  </div>
</DocumentFragment>
`;
